inc/functions.php

```diff
@@ -347,12 +347,12 @@ function preferred_languages_load_textdomain_mofile( $mofile, $domain ) {
 
 	remove_filter( 'load_textdomain_mofile', 'preferred_languages_load_textdomain_mofile' );
 
+	$merge_translations = apply_filters( 'preferred_languages_merge_translations', false, $domain, $current_locale );
+
 	foreach ( $preferred_locales as $locale ) {
 		$preferred_mofile = str_replace( $current_locale, $locale, $mofile );
 
 		if ( is_readable( $preferred_mofile ) ) {
-			$merge_translations = apply_filters( 'preferred_languages_merge_translations', false, $domain, $locale, $current_locale );
```

Comment:
Having the filter inside of the foreach loop gave the illusion merging can be controlled on a per-locale basis, but because of the return statement, this is not happening. This only confuses developers, and I can't really think of any practical use case where this would matter.